crates/bevy_core/src/bytes.rs

```diff
@@ -170,10 +169,8 @@ where
 {
     fn from_bytes(bytes: &[u8]) -> Self {
         unsafe {
-            let byte_ptr = bytes.as_ptr() as *const T;
-            let len = bytes.len() / std::mem::size_of::<T>();
-            let slice = core::slice::from_raw_parts::<T>(byte_ptr, len);
-            slice.to_vec()
+            let (_, body, _) = bytes.align_to::<T>();
```

Comment:
Should there be any assertions that the `head` and `tail` of this tuple are empty slices?

Reply:
I think, like my other comment, that `align_to` is just the wrong approach here.

As a sketch, I think that perhaps it should be more like
```
assert_eq!(bytes.len() % std::mem::size_of::<T>(), 0);
let len = bytes.len() / std::mem::size_of::<T>();
let mut v = Vec::<T>::with_capacity(len);
ptr::copy_nonoverlapping(bytes.as_ptr(), v.as_mut_ptr() as *mut u8, len * std::mem::size_of::<T>());
v.set_len(len);
v
```
because it needs a 1-aligned copy to read from the slice correctly.

Reply:
For `copy_nonoverlapping` doesn't `T` have to be `Copy`?
Currently `T` is only bounded on `Clone`.